src/api/providers/anthropic.ts

```diff
@@ -32,16 +30,19 @@ export class AnthropicHandler implements ApiHandler, SingleCompletionHandler {
 	async *createMessage(systemPrompt: string, messages: Anthropic.Messages.MessageParam[]): ApiStream {
 		let stream: AnthropicStream<Anthropic.Messages.RawMessageStreamEvent>
 		const cacheControl: CacheControlEphemeral = { type: "ephemeral" }
-		const modelId = this.getModel().id
-		const maxTokens = this.getModel().info.maxTokens || 8192
+		let { id: modelId, info: modelInfo } = this.getModel()
+		const maxTokens = modelInfo.maxTokens || 8192
+		const budgetTokens = this.options.anthropicThinking ?? Math.min(maxTokens - 1, 8192)
```

Comment:
I think openrouter had some comment about making this at least 1024 smaller than max tokens

Reply:
https://openrouter.ai/docs/use-cases/reasoning-tokens

Reply:
Oh I see you mention this below - but maybe want better fallback settings? 🤷

Reply:
Right - I think you want to allow for non-thinking output tokens. Will fix.